target/linux/generic/pending-5.10/850-07-net-dsa-qca8k-enable-mtu_enforcement_ingress.patch

```diff
@@ -0,0 +1,31 @@
+From fbd00a0c47c071812bee168fbdb60a447fe752a7 Mon Sep 17 00:00:00 2001
+From: Ansuel Smith <ansuelsmth@gmail.com>
+Date: Mon, 18 Oct 2021 02:47:14 +0200
+Subject: [PATCH 7/7] net: dsa: qca8k: enable mtu_enforcement_ingress
+
+qca8k have a global MTU. Inform DSA of this as the change MTU port
+function checks the max MTU across all port and sets the max value
+anyway as this switch doesn't support per port MTU.
+
+Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
+---
+ drivers/net/dsa/qca8k.c | 3 +++
+ 1 file changed, 3 insertions(+)
+
+diff --git a/drivers/net/dsa/qca8k.c b/drivers/net/dsa/qca8k.c
+index 2491905dafe8..943843c468d1 100644
+--- a/drivers/net/dsa/qca8k.c
++++ b/drivers/net/dsa/qca8k.c
+@@ -1198,6 +1198,9 @@ qca8k_setup(struct dsa_switch *ds)
+ 	/* Set max number of LAGs supported */
+ 	ds->num_lag_ids = QCA8K_NUM_LAGS;
```

Comment:
+@@ -1307,6 +1307,9 @@
+ 	/* Set min a max ageing value supported */
+ 	ds->ageing_time_min = 7000;
+ 	ds->ageing_time_max = 458745000;

Reply:
This patch was not apllying correctly. I fixed it as such.

Reply:
With this patch refresh, all ports are working as expected, wan and lans.

Reply:
Can you put output of ip link? The fix is correct

Reply:
ip link output:
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1502 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:40 brd ff:ff:ff:ff:ff:ff
3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1502 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:3f brd ff:ff:ff:ff:ff:ff
4: lan4@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:3f brd ff:ff:ff:ff:ff:ff permaddr 3c:37:86:13:25:40
5: lan3@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:3f brd ff:ff:ff:ff:ff:ff
6: lan2@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:3f brd ff:ff:ff:ff:ff:ff permaddr 3c:37:86:13:25:40
7: lan1@eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:3f brd ff:ff:ff:ff:ff:ff
8: wan@eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:40 brd ff:ff:ff:ff:ff:ff
16: br-lan: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:3f brd ff:ff:ff:ff:ff:ff
19: ifb4eth0: <BROADCAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc tbf state UNKNOWN mode DEFAULT group default qlen 32
    link/ether 5e:44:c5:9b:e7:73 brd ff:ff:ff:ff:ff:ff
21: wgserver: <POINTOPOINT,NOARP,UP,LOWER_UP> mtu 1420 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/none
22: wlan0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:41 brd ff:ff:ff:ff:ff:ff
23: pppoe-wan: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc tbf state UNKNOWN mode DEFAULT group default qlen 3
    link/ppp
24: wlan1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-lan state UP mode DEFAULT group default qlen 1000
    link/ether 3c:37:86:13:25:42 brd ff:ff:ff:ff:ff:ff
30: ifb4pppoe-wan: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN mode DEFAULT group default qlen 32
    link/ether 66:4b:7c:6d:6a:7f brd ff:ff:ff:ff:ff:ff
```

So far, all my testes are good but I'll report back if I find anything.

Reply:
the mtu to 1502 is suspicious... do you have by luck an old output?

Reply:
How old? I can easily check from my previous dsa based firmware built from your PR #4036. I can also get a log from a vanilla master built with swconfig but that will take a bit longer. Just tell me what you need.

Reply:
nha don't waste time if you already have a log from #4036 it would be good... but don't even try flashing just to get that log. I will investigate if the mtu is correct.

Reply:
In the #4036 PR discussion, there are a number of ip link outputs posted there and they all show mtu 1502.